source/d3d10/runtime_d3d10.cpp

```diff
@@ -279,6 +279,10 @@ void reshade::d3d10::runtime_d3d10::on_present()
 	if (_backbuffer_resolved != _backbuffer)
 		_device->ResolveSubresource(_backbuffer_resolved.get(), 0, _backbuffer.get(), 0, _backbuffer_format);
 
+	// Setup real back buffer
+	auto rtv = _backbuffer_rtv[0].get();
+	_device->OMSetRenderTargets(1, &rtv, nullptr);
+
```

Comment:
Those lines are for fixing Crysis DX10 depth buffer detection

Reply:
But why? As usual I would prefer knowing why this fixes it, since this shouldn't actually do anything.

Reply:
I must admit i don't know myself why those lines fix the db detection for Crysis. If MSAA was enabled, i would understand, but this is not the case. maybe this could be related to motion blur implementation in this specific game....

Reply:
What differs when this is added? Are more draw calls counted (so the correct buffers are already there, just the heuristic picks the wrong one otherwise)? Or are more buffers added to the list?

Reply:
No, the number of depth buffers are still the same. The initialisation seems to be different. When those lines are not present, there is no depth info at all (the displayDepth shader produces a black screen in normal mode).

Reply:
Maybe this pb is related to the dsv creation or the depthstencil creation ?

Reply:
What happens when you unset all render targets there instead?:
```c++
_device->OMSetRenderTargets(0, nullptr, nullptr);
```

Reply:
If that fixes it too, then I think I know what the problem is.
https://docs.microsoft.com/en-us/windows/win32/api/d3d11/nf-d3d11-id3d11devicecontext-pssetshaderresources: ```If an overlapping resource view is already bound to an output slot, such as a rendertarget, then this API will fill the destination shader resource slot with NULL.```
PSSetShaderResources is called first in `render_technique`, which means that if the depth buffer is still bound to the OM (because the game bound it), then it won't actually bound to the ReShade shader.
This also means that you should be able to workaround this by enabling any shader (doesn't matter which one) before DisplayDepth (because ReShade unbinds all render targets after each pass in `render_technique`).